src/mmds/src/lib.rs

```diff
@@ -156,19 +164,28 @@ pub fn parse_request(request_bytes: &[u8]) -> Response {
 #[cfg(test)]
 mod tests {
     extern crate serde_json;
+
     use super::*;
+    use dumbo::tcp::parse_request;
+    use micro_http::{Request, Response, StatusCode, Version};
 
     fn check_http_method_failure(method: String, status: StatusCode, err_msg: String) {
         let request = format!("{} http://169.254.169.254/ HTTP/1.1\r\n\r\n", method);
         let mut expected_response = Response::new(Version::Http11, status);
         expected_response.set_body(Body::new(err_msg));
-        let actual_response = parse_request(request.as_bytes());
+        let actual_response = parse_request(request.as_bytes(), callback);
         assert!(expected_response.status() == actual_response.status());
         assert!(expected_response.body().unwrap() == actual_response.body().unwrap());
     }
 
+    fn check_callback_response(actual: Response, expected: Response) {
+        assert!(expected.status() == actual.status());
+        assert!(expected.body().unwrap() == actual.body().unwrap());
+        assert!(expected.http_version() == actual.http_version());
+    }
```

Comment:
This looks like a `PartialEq` implementation. Would it make sense to `impl PartialEq` here instead?

Reply:
@lauralt Thank you for review! That makes sense. I added `#[derive(PartialEq)]` for the necessary part and remove `check_callback_response` function.